File: Easy/Problem1.java
import java.util.Arrays;

public class Problem1 {
    public int[] twoSum(int[] nums, int target) {
        int[] ans = new int[2];
        for (int i = 0 ; i < nums.length ; i++) {            
            boolean flag = false; 
            for (int j = 0 ; j < nums.length ; j++)
                if (j != i && nums[i] + nums[j] == target) {
                    if (i > j) {
                        ans[0] = j;
                        ans[1] = i;
                    }else {
                        ans[0] = i;
                        ans[1] = j;
                    }
                    flag = true;
                    break;
                }
            if (flag) break;    
        } 
        return ans;
    }
}

File: Easy/Problem1.cpp
#include <iostream>
#include <vector>
#include <map>

using namespace std;

vector<int> twoSum(vector<int>& nums, int target) {
    map<int , int>m;
    for (int i = 0 ; i < nums.size() ; i++) m[nums[i]]++;
    int idxa = 0 , idxb = 0;
    for (int i = 0 ; i < nums.size() ; i++) {
        if (target - nums[i] == nums[i] && m[nums[i]] >= 2) {
            idxa = i;
            break;
        }
        if (m[target - nums[i]] > 0 && target - nums[i] !=  nums[i]) { //3 2 4
            idxa = i;
            break;
        }
    }
    for (int i = 0 ; i < nums.size() ; i++) {
        if (i != idxa && nums[i] == target - nums[idxa]) {
            idxb = i;
            break;
        }
    }
    vector<int> v(2);
    v[0] = idxa;v[1] = idxb;
    return v;
}
